src/solvers/dgmulti/flux_differencing.jl

```diff
@@ -413,10 +413,10 @@ end
 # Computes flux differencing contribution from each Cartesian direction over a single element.
 # For dense operators, we do not use sum factorization.
 @inline function local_flux_differencing!(fluxdiff_local, u_local, element_index,
```

Comment:
Did you run some performance tests? Passing a `function` through this method may let Julia avoid specializing on it, leading to some allocations. If so, we should force specialization on `volume_flux` to be sure.

Reply:
Good point. I added this for more generality, but forgot about https://docs.julialang.org/en/v1/manual/performance-tips/#Be-aware-of-when-Julia-avoids-specializing. 

I'll check the timings. Is there a potential downside to specializing `volume_flux`? Does it add to precompilation time?

Reply:
Timings are a bit inconsistent if I just run the solver. Using `@btime`, I get the following:

On `main`:
```julia
julia> u = sol.u[end]; du = similar(u); @btime Trixi.rhs!($du, $u, $semi, $0.0)
  416.383 μs (4108 allocations: 197.58 KiB)
```
On this PR with `volume_flux` specialization
```julia
julia> u = sol.u[end]; du = similar(u); @btime Trixi.rhs!($du, $u, $semi, $0.0)
  436.073 μs (6156 allocations: 373.44 KiB)
```
On this PR without `volume_flux` specialization
```julia
julia> u = sol.u[end]; du = similar(u); @btime Trixi.rhs!($du, $u, $semi, $0.0)
  434.624 μs (6156 allocations: 373.44 KiB)
```
So somehow the allocations have increased in this PR, but this seems unrelated to specialization.

Reply:
The performance regression seems to happen in 1c268e2d2df399fea1cb3e9c414c99782304c51b

Reply:
Inlining appears to fix this in 7c4b1cc47dfbe80f06567eeee80a66ac6fccc284. 
```julia
julia> u = sol.u[end]; du = similar(u); @btime Trixi.rhs!($du, $u, $semi, $0.0)
  364.882 μs (4108 allocations: 197.44 KiB)
```

@ranocha if I specialize `volume_flux`, it doesn't result in any change in allocations compared with 7c4b1cc47dfbe80f06567eeee80a66ac6fccc284. Do you want to keep the specialization or should we remove it?

Reply:
We can keep it if it's fine. However, `4108 allocations: 197.44 KiB` also looks like more I would have expected?

Did you test Julia v1.8 and v1.9?

Reply:
Yeah, these timings were on 1.9. I see roughly the same number of allocations on 1.8.5 as well.
```julia
julia> u = sol.u[end]; du = similar(u); @btime Trixi.rhs!($du, $u, $semi, $0.0)
  1.295 ms (4096 allocations: 192.00 KiB)
```
I still see some allocations in https://github.com/trixi-framework/Trixi.jl/issues/1435 even after your PR, so maybe some of these are related to what we're seeing here?

Reply:
If you're OK with it, I can address the allocations in a separate PR.

Reply:
Sure :+1: